database/postgres/dal/preferences.py

```diff
@@ -0,0 +1,87 @@
+from database.models import UserPreference, Sport
+import database.models
+from dto.api_input import TablesAndColumnsDTO
+from service.api_logic.sports_logic import get_all_sports
+
+
+class PreferencesDAL:
+
+    def __init__(self, session=None):
+        self.session = session
+
+
+    def getattr_tables_and_columns_by_type(self, type_dto):
+        main_table = getattr(database.models, type_dto.main_table, None)
+        related_table = getattr(database.models, type_dto.related_table, None)
+        user_id_field = getattr(main_table, type_dto.user_id_field, None)
+        type_id_field = getattr(main_table, type_dto.type_id_field, None)
+        related_name = getattr(related_table, type_dto.related_name, None)
+        related_logo = getattr(related_table, type_dto.related_logo, None)
+        related_id = getattr(related_table, type_dto.related_id, None)
+
+        return TablesAndColumnsDTO(
+            main_table=main_table,
+            related_table=related_table,
+            user_id_field=user_id_field,
+            type_id_field=type_id_field,
+            related_name=related_name,
+            related_logo=related_logo,
+            related_id=related_id
+        )
+
+
+    def get_all_sport_preferences(self):
+        return get_all_sports()
```

Comment:
your DAL should not depend on BL layer.

Basically, you can't use neither API nor BL inside your database project